apps/backend/src/mutations/InstrumentMutations.ts

```diff
@@ -132,74 +133,93 @@ export default class InstrumentMutations {
-        { args }
-      );
-    }
+      if (!allProposalsAreOnSameCallAsInstrument) {
+        return rejection(
+          'One or more proposals can not be assigned to instrument, because instrument is not in the call',
+          { args }
+        );
+      }
 
-    const instrument = await this.dataSource.getInstrument(args.instrumentId);
+      const instrument = await this.dataSource.getInstrument(instrumentId);
 
-    if (!instrument) {
-      return rejection(
-        'Cannot assign the proposal to the instrument because the proposals call has no such instrument',
-        { agent, args }
-      );
-    }
+      if (!instrument) {
+        return rejection(
+          'Cannot assign the proposal to the instrument because the proposals call has no such instrument',
+          { agent, args }
+        );
+      }
 
-    const proposalPks = args.proposals.map((proposal) => proposal.primaryKey);
+      const proposalPks = args.proposals.map((proposal) => proposal.primaryKey);
 
-    for await (const proposalPk of proposalPks) {
-      const technicalReview = await this.reviewDataSource.getTechnicalReview(
-        proposalPk
-      );
+      for await (const proposalPk of proposalPks) {
+        const technicalReview =
+          await this.reviewDataSource.getProposalInstrumentTechnicalReview(
+            proposalPk,
+            instrumentId
+          );
 
-      if (technicalReview) {
-        await this.proposalDataSource.updateProposalTechnicalReviewer({
-          userId: instrument.managerUserId,
-          proposalPks: [proposalPk],
-        });
-      } else {
-        await this.reviewDataSource.setTechnicalReview(
-          {
-            proposalPk: proposalPk,
-            comment: null,
-            publicComment: null,
-            reviewerId: instrument.managerUserId,
-            timeAllocation: null,
-            status: null,
-            files: null,
-            submitted: false,
-          },
-          false
-        );
-        await this.proposalDataSource.updateProposalTechnicalReviewer({
-          userId: instrument.managerUserId,
-          proposalPks: [proposalPk],
-        });
+        if (technicalReview) {
+          await this.proposalDataSource.updateProposalTechnicalReviewer({
+            userId: instrument.managerUserId,
+            proposalPks: [proposalPk],
+            instrumentId: instrument.id,
+          });
+        } else {
+          await this.reviewDataSource.setTechnicalReview(
+            {
+              proposalPk: proposalPk,
+              comment: null,
+              publicComment: null,
+              reviewerId: instrument.managerUserId,
+              timeAllocation: null,
+              status: null,
+              files: null,
+              submitted: false,
+              instrumentId: instrument.id,
+            },
+            false
+          );
+          await this.proposalDataSource.updateProposalTechnicalReviewer({

```

Comment:
Why do we need to update proposal Technical reviewer here? Is this in case the proposal was added to another instrument, and reassigned to different, then the revierwer needs to change?

Reply:
Yes and we need some initial value for the reviewerId and I think instrument manager works the best in this case. This is changed when the real review is saved or submitted then the reviewer gets updated to the one that did the last save/submit.